### PR TITLE
feat(mojaloop/#2092)!: prefixes for api definitions and upgraded node version

### PR DESCRIPTION
**Pull request by vijayg10**
feat(mojaloop/[#2092](https://github.com/mojaloop/project/issues/2092)): upgrade nodeJS version for core services - https://github.com/mojaloop/project/issues/2092
- Upgrade to node v16
- Update packages/linting/audit/dockerfile
- Update ci with dep/audit checks
- Installed pre-commit hooks
- feat:added options to api management page

Notes:
- Helm chart mountPaths need to be updated from `/opt/mojaloop-testing-toolkit-ui` to `/opt/app` as follows:
    ```YAML
        volumeMounts:
        - mountPath: /opt/app/config
          name: <deployment-name>
    ```

BREAKING CHANGE: Major version bump for node v16 LTS support, re-structuring of project directories to align to core Mojaloop repositories and docker image now uses `/opt/app` instead of `/opt/mojaloop-testing-toolkit-ui` which will impact config mounts.

Major version bump since this is a big upgrade.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,23 +15,28 @@
   "homepage": "https://localhost:6060/",
   "scripts": {
```
Comment (mdebarros) [inline package.json:16]: 1. Please add  the following:
  a. Contributors: https://github.com/mojaloop/simulator/blob/master/package.json#L8
  b. Engines: https://github.com/mojaloop/simulator/blob/master/package.json#L20
2. Why is the license MIT? I assume this should be apache2.0? https://github.com/mojaloop/simulator/blob/master/package.json#L7

**Review by mdebarros — COMMENTED** [review]
Please some comments.

In addition:
- can you align the folder structure to the general pattern we follow at Mojaloop for consistency, please? I.e. /src, /test etc at root level.
- please add CODEOWNERS as well, [ref](https://github.com/mojaloop/email-notifier/pull/202/files#diff-fcf14c4b7b34fe7a11916195871ae66a59be87a395f28db73e345ebdc828085b).
- Please add the following to the end of your README.MD
  a. auditing-dependencies: https://github.com/mojaloop/central-ledger/blob/master/README.md#auditing-dependencies
  b. container-scans: https://github.com/mojaloop/central-ledger/blob/master/README.md#container-scans
  c. automated-releases: https://github.com/mojaloop/central-ledger/blob/master/README.md#automated-releases
  
Take note that you should create a snapshot for testing on the OSS environment before any merges are done here, example PR where this is done: https://github.com/mojaloop/als-oracle-pathfinder/pull/72

File: Dockerfile [hunk 2]
```diff
@@ -1,28 +1,42 @@
...
     && cd $(npm root -g)/npm \
     && npm config set unsafe-perm true \
     && npm install -g node-gyp
 
-COPY package.json package-lock.json* /opt/mojaloop-testing-toolkit-ui/
-RUN npm install --silent
-#RUN npm install react-scripts@3.0.1 -g --silent
+COPY package.json package-lock.json* /opt/app/
+
+RUN npm install
```
Comment (mdebarros) [inline Dockerfile:11]:
```suggestion
RUN npm ci
```

File: .ncurc.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+reject: [
+  # TODO: Incompatibility issues found with the TTK backend if we update socket.io client
+  "socket.io-client",
+  # TODO: Upgrading react may cause many implications
```
[the remark this answers is not in the document]
Comment (vijayg10) [reply .ncurc.yaml:5]: Yes I tried to upgrade and got many issues. Spent some time and not able to fix them as the difference in the react library functions. I added a more elaborated comment now.

File: audit-ci.jsonc [hunk 4]
```diff
@@ -0,0 +1,55 @@
+{
+  "$schema": "https://github.com/IBM/audit-ci/raw/main/docs/schema.json",
+  // audit-ci supports reading JSON, JSONC, and JSON5 config files.
+  // Only use one of ["low": true, "moderate": true, "high": true, "critical": true]
+  "moderate": true,
+  "allowlist": [
+    // These are the exceptions for react, react-scripts and their dependencies as it is very hard to update react version at the moment.
```
[the remark this answers is not in the document]
Comment (vijayg10) [reply audit-ci.jsonc:7]: Those are all react components and its dependencies.

File: .circleci/config.yml [hunk 5]
```diff
@@ -368,36 +683,118 @@ workflows:
...
               ignore:
                 - /.*/
-
+      - publish-snapshot:
+          context: org-global
+          requires:
+            - pr-tools/pr-title-check
+            - test-dependencies
+            - test-lint
+            # - test-unit
```
[the remark this answers is not in the document]
Comment (vijayg10) [reply .circleci/config.yml:788]: Done

File: README.md [hunk 6]
```diff
@@ -54,3 +54,8 @@ Please follow this link to learn about backend service.
 
 https://github.com/mojaloop/ml-testing-toolkit/blob/master/README.md
 
+-------
+
+a. auditing-dependencies: https://github.com/mojaloop/central-ledger/blob/master/README.md#auditing-dependencies
```
Comment (mdebarros) [inline README.md:59]: What I meant by this is that you should copy and paste those sections into this document, not reference as you have done.

**Review by mdebarros — COMMENTED** [review]
Hey @vijayg10,

One minor comment left for you to address.

**Comment by mdebarros** [comment]
@vijayg10 I also updated your PR description, please take note of it, and ensure that you include the PR description as part of your merge commit message.

We need to be specific about what is breaking! Node upgrade is nice, but the real breaking change here is the dockerfile working directory that needs to be explicitly called out. Unless there is something else that you are aware of that needs to be added.

File: README.md (hunk 6 above)
Comment (vijayg10) [reply README.md]: Sorry for being so dumb :). Done

**Comment by mdebarros** [comment]
I further update the PR description. See the changes above.

**Review by mdebarros — APPROVED** [review]
+1

**Pull request merged**